pretix_oidc/auth.py

```diff
@@ -39,15 +41,34 @@ def __init__(self):
             )
 
             self.client = Client(client_authn_method=CLIENT_AUTHN_METHOD)
+            if not config.get("oidc", "skip_provider_discovery", fallback=False):
+                # If skip_provider_discovery is set, we do not fetch the provider config
+                # but use the provided information directly.
+                self.client.provider_config(op_info["issuer"])
+            self.client.handle_provider_config(op_info, op_info["issuer"])
+
+            missing_endpoints = {
+                "authorization_endpoint",
+                "token_endpoint",
+                "userinfo_endpoint",
+                "end_session_endpoint"
+            } - set(
+                {k:v for k,v in self.client.__dict__.items() if k.endswith("_endpoint") and v is not None}.keys()
```

Comment:
If we're interested in the keys only wouldn't `{k for k,v in…}` work on its own?